src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_validators.py

```diff
@@ -0,0 +1,241 @@
+
+
+def process_certificate_cancel_namespace(namespace):
+    namespace.cancellation_requested = True
+
+
+def process_secret_set_namespace(namespace):
+
+    validate_tags(namespace)
+
+    content = namespace.value
+    file_path = namespace.file_path
+    encoding = namespace.encoding
+    tags = namespace.tags or {}
+
+    use_error = CLIError("incorrect usage: [Required] --value VALUE | --file PATH")
+
+    if (content and file_path) or (not content and not file_path):
+        raise use_error
+
+    encoding = encoding or 'utf-8'
+    if file_path:
+        if encoding in secret_text_encoding_values:
+            with open(file_path, 'r') as f:
+                try:
+                    content = f.read()
+                except UnicodeDecodeError:
+                    raise CLIError("Unable to decode file '{}' with '{}' encoding.".format(
+                        file_path, encoding))
+                encoded_str = content
+                encoded = content.encode(encoding)
+                decoded = encoded.decode(encoding)
+        elif encoding == 'base64':
+            with open(file_path, 'rb') as f:
+                content = f.read()
+                try:
+                    encoded = base64.encodebytes(content)
+                except AttributeError:
+                    encoded = base64.encodestring(content) # pylint: disable=deprecated-method
+                encoded_str = encoded.decode('utf-8')
+                decoded = base64.b64decode(encoded_str)
+        elif encoding == 'hex':
+            with open(file_path, 'rb') as f:
+                content = f.read()
+                encoded = binascii.b2a_hex(content)
+                encoded_str = encoded.decode('utf-8')
+                decoded = binascii.unhexlify(encoded_str)
+
+        if content != decoded:
+            raise CLIError("invalid encoding '{}'".format(encoding))
+
+        content = encoded_str
+
+    tags.update({'file-encoding': encoding})
+    namespace.tags = tags
+    namespace.value = content
+
+# PARAMETER NAMESPACE VALIDATORS
+
+
+def get_attribute_validator(name, attribute_class, create=False):
+
+    def validator(ns):
+        ns_dict = ns.__dict__
+        enabled = not ns_dict.pop('disabled') if create else ns_dict.pop('enabled')
+        attributes = attribute_class(
+            enabled,
+            ns_dict.pop('not_before'),
+            ns_dict.pop('expires'))
+        setattr(ns, '{}_attributes'.format(name), attributes)
+
+    return validator
+
+
+def validate_key_import_source(ns):
+    byok_file = ns.byok_file
+    pem_file = ns.pem_file
+    pem_password = ns.pem_password
+    if (not byok_file and not pem_file) or (byok_file and pem_file):
+        raise ValueError('supply exactly one: --byok-file, --pem-file')
+    if byok_file and pem_password:
+        raise ValueError('--byok-file cannot be used with --pem-password')
+    if pem_password and not pem_file:
+        raise ValueError('--pem-password must be used with --pem-file')
```

Comment:
It looks like this code was copy-pasted in from the keyvault module.  You should only copy what you actually need. I believe this is dead code.